server/stream.go

```diff
@@ -1444,6 +1467,13 @@ func (mset *stream) updateWithAdvisory(config *StreamConfig, sendAdvisory bool)
 		return NewJSStreamInvalidConfigError(err, Unless(err))
 	}
 
+	jsa.mu.Lock()
```

Comment:
This an RW? Can we use RLock?

Reply:
Yes, we can use RLock here I will change.   subjectsOverlap() is called under Lock but that is a side-effect of single caller's RW critical section.